File: android/core/network/src/main/kotlin/com/aibook/android/core/network/api/BookApi.kt
package com.aibook.android.core.network.api

import com.aibook.android.core.network.api.dto.BookDTO
import com.aibook.android.core.network.api.dto.BookPage
import com.aibook.android.core.network.api.dto.ProcessedContentResponse
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query
import okhttp3.ResponseBody
import retrofit2.http.Streaming

interface BookApi {
    @GET("api/books")
    suspend fun getBooks(
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20,
        @Query("sortBy") sortBy: String = "createdAt",
        @Query("sortDir") sortDir: String = "desc",
        @Query("format") format: String? = null,
        @Query("status") status: String? = null
    ): BookPage

    @GET("api/books/search")
    suspend fun searchBooks(
        @Query("keyword") keyword: String,
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20
    ): BookPage

    @GET("api/books/favorites")
    suspend fun getFavoriteBooks(
        @Query("page") page: Int = 0,
        @Query("size") size: Int = 20
    ): BookPage

    @GET("api/books/{id}")
    suspend fun getBookById(@Path("id") id: Long): BookDTO

    @POST("api/books/{id}/open")
    suspend fun recordBookOpen(
        @Path("id") id: Long,
        @Query("versionId") versionId: Long? = null
    )

    @Streaming
    @GET("api/books/{id}/content")
    suspend fun getBookContent(
        @Path("id") id: Long,
        @Query("versionId") versionId: Long? = null
    ): ResponseBody

    @GET("api/books/{id}/content-processed")
    suspend fun getProcessedContent(@Path("id") id: Long): ProcessedContentResponse
}

File: ios/AiBook/Core/Network/BookApi.swift
import Foundation

// MARK: - BookApi（书籍 API，与安卓 BookApi 对齐）

struct BookListResponse: Decodable, Sendable {
    let content: [BookDto]
    let totalElements: Int
    let totalPages: Int
    let number: Int
}

struct BookDto: Decodable, Sendable {
    let id: Int64
    let title: String
    let author: String?
    let format: String?
    let coverUrl: String?
    let description: String?
    let fileSize: Int64?
    let pageCount: Int?
}

struct ReadingProgressRequest: Encodable, Sendable {
    let chapterIndex: Int
    let chapterTitle: String?
    let progressPercent: Double
}

struct ReadingTimeRequest: Encodable, Sendable {
    let readingTimeSeconds: Int
}

@MainActor
final class BookApi {
    private let client: ApiClient

    init(client: ApiClient) {
        self.client = client
    }

    func getBooks(page: Int = 0, size: Int = 20, sort: String = "importedAt,desc") async throws -> BookListResponse {
        try await client.get("api/books?page=\(page)&size=\(size)&sort=\(sort)")
    }

    func searchBooks(query: String, page: Int = 0, size: Int = 20) async throws -> BookListResponse {
        let encoded = query.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? query
        return try await client.get("api/books/search?keyword=\(encoded)&page=\(page)&size=\(size)")
    }

    func getFavorites(page: Int = 0, size: Int = 20) async throws -> BookListResponse {
        try await client.get("api/books/favorites?page=\(page)&size=\(size)")
    }

    func getBookDetail(id: Int64) async throws -> BookDto {
        try await client.get("api/books/\(id)")
    }

    func getProcessedContent(id: Int64) async throws -> String {
        let data = try await client.requestData("api/books/\(id)/content-processed")
        return String(data: data, encoding: .utf8) ?? ""
    }

    func saveReadingProgress(bookId: Int64, progress: ReadingProgressRequest) async throws {
        let _: EmptyResponse = try await client.post("api/books/\(bookId)/reading-progress", body: progress)
    }

    func updateReadingTime(bookId: Int64, seconds: Int) async throws {
        let _: EmptyResponse = try await client.put("api/books/\(bookId)/reading-time", body: ReadingTimeRequest(readingTimeSeconds: seconds))
    }
}
